torch/testing/_internal/optests/compile_check.py

```diff
@@ -0,0 +1,74 @@
+import torch
+from .make_fx import make_fx_check
+from .aot_autograd import aot_autograd_check
+from .common import TestFrameworkError
+from torch._subclasses.schema_check_mode import SchemaCheckMode
+
+
+def compile_check(
+        func,
+        args,
+        kwargs,
+        *,
+        dynamic_only=False,
+        inference_only=False,
+        fullgraph=True,
+        raise_error=True):
+    """Check if torch.compile supports a function.
+
+    Args:
+        func (function): a Python function that takes at least one Tensor
+            as input and returns a Tensor or a Tuple of Tensors.
+        args (Tuple): args to the function
+        kwargs (dict): kwargs to the function
+        dynamic_only (bool, optional): If the function only works with dynamic
+            shapes. This can happen if it returns Tensors whose shape are
+            dependent on the data on the input Tensors. If True, we skip
+            tests related to torch.compile with static shapes.
+        inference_only (bool, optional): If the function only works with
+            inputs that do not require grad. If True, we will skip
+            autograd-related tests.
+        fullgraph (bool, optional): If we expect the entire function
+            to be captured with torch.compile without any graph breaks.
+
+    """
+    def run_static_or_dynamic_tests(dynamic):
+        tracing_mode = 'symbolic' if dynamic else 'fake'
+        with ignore_test_framework_error():
```

Comment:
Maybe make test framework fail louder? If a user tries to test something and runs into a test framework error, it would appear as though as the test passed right?